pkg/controllers/status/work_status_controller.go

```diff
@@ -251,10 +256,27 @@ func (c *WorkStatusController) syncWorkStatus(key util.QueueKey) error {
 		// status changes.
 	}
 
+	if reason, ready := c.isReflectStatusReady(observedObj); !ready {
+		// Just return nil and no need to reflect the status, waiting for the next correct event.
```

Comment:
How about this comment: When the generation in the resource does not reach a consistent state, skip reflect status and wait for the next update event.